akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -104,6 +104,16 @@ object ShardCoordinator {
         rebalanceInProgress: Set[ShardId]): Future[Set[ShardId]]
   }
 
+  /**
+   * Shard allocation strategy where start is called by the shard coordinator before any calls to
+   * rebalance or allocate shard. This can be used if there is any expensive initialization to be done
+   * that you do not want to to in the constructor as it will happen on every node rather than just
+   * the node that hosts the ShardCoordinator
+   */
+  trait StartableAllocationStrategy extends ShardAllocationStrategy {
+    def start(): Unit
```

Comment:
I don't think we need to have the shard coordinator delay its startup for this.
If an implementation needs to do something async it can chain the futures returned from rebalance and allocate

Reply:
Alright, maybe worth mentioning in the docs then given that this is public API?